SwiftIntro.playground/Pages/Untitled Page.xcplaygroundpage/Contents.swift

```diff
@@ -228,49 +303,67 @@ var arrayOfItems: [Item] = [
 ]
 
 //create items names array of Strings from arrayOfItems use map is possible
+var namesOfItems: [String] = arrayOfItems.map { $0.name }
+namesOfItems.forEach { print($0) }
 
-//var namesOfItems: [String] =
-
-//namesOfItems.forEach { print($0) }
-
+//MARK: - NEED HELP >>>>>>>>>>
 //Create function that transforms value from dictionary into array of strings - description about movie (Good > 5, bad <=5
 //["Home Alone 1 (Good)", "Star Wars (Bad)", "Harry Potter (Good)", "Home Alone 4 (Bad)"]
 
 let movies: [String: Float] = ["Home Alone 1": 10, "Star wars": 5, "Harry Potter": 8, "Home Alone 4": 2]
 
-//let mappedMovies: [String]
+func transformValueToString(dict: [String: Float]) {
+//    if (dict.map { $0.value }) > 5 {
+//
+//    } else if (dict.map { $0.value }) <= 5 {
+//
+//    }
+}
+
+var mappedMovies: [String]
 //mappedMovies.forEach { print($0) }
+//mappedMovies
+//MARK: - NEED HELP <<<<<<<<<<
 
+//filter
 let arrayOfIntegers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
 
 var newArray: [Int] = []
 
 //Rewrite it using filter
-arrayOfIntegers.forEach {
-    if $0 % 2 == 0 {
-        newArray.append($0)
-    }
-}
+//arrayOfIntegers.forEach {
+//    if $0 % 2 == 0 {
+//        newArray.append($0)
+//    }
+//}
+newArray.append(contentsOf: arrayOfIntegers.filter { ($0 % 2) == 0 })
 
 newArray.forEach { print($0) }
+newArray
 
 //flat map
-
 var arrayOfArrays: [[Int]] = [[1], [2], [1, 4, 6], [6, 5, 3], [2,5,6]]
-
-//arrayOfArrays.forEach { print($0) }
+arrayOfArrays.forEach { print($0) }
 
 //create array of Integers, from given array - arrayOfArrays which has all values from previous array
-
-//let arrayOfInt: [Int]
-//arrayOfInt.forEach { print($0) }
+let arrayOfInt: [Int] = arrayOfArrays.flatMap { $0 }
+arrayOfInt.forEach { print($0) }
+arrayOfInt
 
 //remove all nil values from given array, use compactMap if possible
 var arrayOfRandomObjects: [Any?] = [1, "aasd", 2, "fasf", nil, nil, "value"]
+var filteredArrayOfRandomObjects: [Any] = []
 
-//arrayOfRandomObjects.forEach {
-//    print("\($0)")
-//}
+for optionalElement in arrayOfRandomObjects {
+    if let unwrappedElement = optionalElement {
+        filteredArrayOfRandomObjects.append(unwrappedElement)
+    }
+}
```

Comment:
try to check slide no. 22, there is a much easier way to do it :)

Reply:
let arrayWithNoNils = arrayOfRandomObjects.compactMap { $0 }